src/importmesh.cpp

```diff
@@ -130,6 +130,13 @@ bool LinkStl(const Platform::Path &filename, EntityList *el, SMesh *m, SShell *s
     char str[80] = {};
     f.read(str, 80);
     
+    if(str[0]=='s' && str[1]=='o' && str[2]=='l' && str[3]=='i' && str[4]=='d') {
```

Comment:
I love this! :-D
C++ be damned, STL be damned even C and `strstr` be damned; we're just gonna check the bytes one by one! ;-)

In fact I am not even entirely joking, enen though I did write `strstr` EXACTLY on this line in my local copy [when we discussed the bug originally](https://github.com/solvespace/solvespace/issues/1084#issuecomment-896969284). Then I decided I would rather write text format STL parsing... and never got to it (so far) :-(

In fact I would almost merge this :-)

Reply:
@ruevs We really can't use standard string functions here because str may not be null terminated. I spent a few minutes looking for a standard comparison that would work and just decided that this one liner is self explanatory and simple enough ;-)

I'm more concerned about the useless error message it produces. Making a dialog for that doesn't seem great either since the "correct" thing to do is handle the text STL file, but that will still need this check.

Reply:
I did do
```C++
char str[81] = {}
```
 to guarantee the null termination for `strstr` ;-).

 But anyway I totally agree with everything you wrote.

Reply:
You should have submitted a PR...

Reply:
@ruevs  look at the first function in file.cpp   I think it fits the bill.

Reply:
It does not, because of `strlen` but `if(0==memcmp("solid", str, 5))` should. I should have thought of that :-) (writing on the phone, so not tested)

Reply:
OK added a commit with `memcmp`.

It does not matter at all but it optimizes beautifully :-)
![image](https://user-images.githubusercontent.com/15338069/150748597-fb72672c-8b57-4ac0-ba15-25732c5615ab.png)

Reply:
StrStartsWith only uses strlen on the second string, which would be "solid" and would be terminated in this case.